packages/trader/src/AppV2/Containers/Chart/__tests__/contract-details-chart.spec.tsx

```diff
@@ -0,0 +1,35 @@
+import React from 'react';
+import { render, screen } from '@testing-library/react';
+import ContractDetailsChart from '../contract-details-chart';
+import { mockStore } from '@deriv/stores';
+import TraderProviders from '../../../../trader-providers';
+import { Router } from 'react-router';
+import { createBrowserHistory } from 'history';
+
+jest.mock('Modules/SmartChart', () => ({
+    SmartChart: () => <div data-testid='dt_mock_chart'>Mocked Chart</div>,
+}));
+jest.mock('react-router-dom', () => ({
+    useLocation: jest.fn(() => ({
+        pathname: '/',
+    })),
+    withRouter: jest.fn(children => <div>{children}</div>),
+}));
+
+describe('Contract Replay Chart', () => {
+    it('should render the chart correctly', () => {
+        const store = mockStore({});
+        const history = createBrowserHistory();
+
+        render(
+            <Router history={history}>
+                <TraderProviders store={store}>
+                    <ContractDetailsChart />
+                </TraderProviders>
+            </Router>
+        );
+        const mockChartElement = screen.getByTestId('dt_mock_chart');
+        expect(mockChartElement).toBeInTheDocument();
+        expect(screen.getByText('Mocked Chart')).toBeInTheDocument();
```

Comment:
Do we need to check twice? First you've checked data-testid, then text of the same element